pip_check_reqs/common.py

```diff
@@ -183,21 +189,23 @@ def is_package_file(path):
     return ""
 
 
-def ignorer(ignore_cfg):
+def ignorer(ignore_cfg: List[str]) -> Callable[..., bool]:
     if not ignore_cfg:
         return lambda candidate: False
 
-    def f(candidate, ignore_cfg=ignore_cfg):
+    def f(
+        candidate: Union[str, ParsedRequirement],
+        ignore_cfg: List[str] = ignore_cfg,
+    ) -> bool:
         for ignore in ignore_cfg:
-            try:
-                candidate_path = install_req_from_line(
+            if isinstance(candidate, str):
+                candidate_path = candidate
+            else:
+                optional_candidate_path = install_req_from_line(
                     candidate.requirement,
```

Comment:
I don't know what type `candidate` is here and we have no tests which have a `candidate.requirement`.

Therefore, there is a risk that a user will hit this as a bug. I think that's the risk we take, and we'll fix it if we hear reports.

cc @MrMino